pkg/syncer/syncer.go

```diff
@@ -453,6 +461,12 @@ func transformName(syncedObject *unstructured.Unstructured, direction SyncDirect
 		if syncedObject.GetKind() == "ServiceAccount" && syncedObject.GetName() == "default" {
 			syncedObject.SetName("kcp-default")
 		}
+		// TODO(jmprusi): We are rewriting the name of the object into a non random one so we can reference it from the deployment transformer
+		//                but this means that means than more than one default-token-XXXX object will overwrite the same "kcp-default-token"
+		//				  object. This must be fixed.
+		if syncedObject.GetKind() == "Secret" && strings.Contains(syncedObject.GetName(), "default-token-") {
```

Comment:
make all of these fully-qualified group-version-kind comparisons. There could be foo/v1 Secrets too.